app/api/scorecard.py

```
@@ -125,3 +125,205 @@ def deleteScorecard(scorecardID: int,
+                criteria=metric.criteria,
+                type=metric_type_map[metric.id]  # Add the type directly from retrieved metric
+            )
+            objects.append(metric)
+
+        check_metric(objects)
+        
+        if (update_scorecard.metrics):
+            # Delete old metrics related to this scorecard
+            scorecardMetricsCrud.deleteByScorecardId(scorecardID)
+            # Create new ones
+            for metric in update_scorecard.metrics:
+                metric.desiredValue = stringify_value(metric.desiredValue)
+                metricUpdated = schemas.ScoreCardMetricsCreate(
+                scoreCardId=scorecardID,
+                metricId=metric.id,
+                criteria=metric.criteria,
+                weight=metric.weight,
+                desiredValue=metric.desiredValue
+                )
+                scorecardMetricsCrud.create(metricUpdated)
+
+    
+    #SERVICE UPDATE 
+    if update_scorecard.services:
+        # if the scorecardids are passed duplicated
+        servicesIDsSet = list(set(update_scorecard.services))
+        # i need to check if the serviceid is found or not
+        services = serviceCrud.getByServiceIds(servicesIDsSet)
+        if (len(services) != len(servicesIDsSet)):
+            raise HTTPResponseCustomized(status_code=400, detail="Service not found")
+        
+        if (update_scorecard.services):
+            # Delete old services related to this scorecard
+            scorecardServiceCrud.deleteByScorecardId(scorecardID)
+            # Create new ones
+            for serviceID in update_scorecard.services:
+                serviceScorecard = schemas.MicroserviceScoreCardUpdate(
+                    scoreCardId=scorecardID,
+                    microserviceId=serviceID
+                )
+                try:
+                    scorecardServiceCrud.create(serviceScorecard)
+                except Exception as e:
+                    return e
```

should we return the error, re-raise it, or re-raise our custom HTTPException ?